### ISSUE TITLE
Check existance of $HOME/bin before adding to PATH

### ISSUE THREAD

**Issue by bohni** [git-for-windows/git#5294]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

It's just an optimization for a *nix shell script. The behaviour is still present in current main branch. 

https://github.com/git-for-windows/build-extra/blob/6735402549b1776f65723e129b9208fa0ee5056b/git-extra/env.sh#L2

```
export PATH="$HOME/bin:$PATH"
```

the PATH should only be extended, if the directory exists and is not already included in PATH

```
if [ -d ${HOME}/bin ]
then
  if [[ ! $PATH = *${HOME}/bin* ]]
  then
    export PATH="$HOME/bin:$PATH"
  fi
fi
```

Maybe the second test (that PATH already includes the directory) must be omitted because not every shell supports this check.

**Comment by dscho** [comment]
I would consider this a feature rather than a bug, as non-existing components should be ignored, but you can always create the `~/bin` folder later and it would be picked up automatically.

What is the exact problem you are trying to solve?

**Comment by bohni** [comment]
Yes, it's a feature request, not a bug. But github only has "new issue"...

The reason is, that I spend quite a while searching my own startup scripts (not only bash, because I'm in this case on windows), looking for the place, why this directory is in my `PATH` variable and I didn't find it there, because in these scripts that check is used...

I based my bash scripts on the debian `.profile` (https://sources.debian.org/src/bash/4.3-11/debian/skel.profile/) and expanded it with this second check.

**Comment by dscho** [comment]
Okay, I understand the intention now.

**Comment by dscho** [comment]
@bohni do you plan on working on this? If not, I'd like to close this ticket.

**Comment by bohni** [comment]
No completed, but won't fix
